### PR TITLE
19 replace const generics with attributes

### PR DESCRIPTION
**Pull request by simonengstrm**
Refactores the code to use attributes instead of const generics.

Note that some of the structs used in the proptests now look redundant in their definitions. Should it be kept for structure or can we simplify it?
image

Solves #19

### CODE REVIEW

File: src/register.rs [hunk 1]
```diff
@@ -20,13 +20,14 @@ pub struct Register<const N: usize> {
...
+    size: usize
 }
 
 
-impl<const N: usize> Register<N> {
+impl Register {
 
     /// Creates a new state with an array of booleans with size N 
-    pub fn new(input_bits: [bool; N]) -> Self {
+    pub fn new(input_bits: Vec<bool>) -> Self {
```
Comment (BAMF0) [inline src/register.rs:30]: This could be replaced with a slice `&[bool]`

File: tests/register_operation_tests.rs [hunk 2]
```diff
@@ -58,7 +58,7 @@ proptest!(
         state7 in any::<bool>(),
         state8 in any::<bool>()
     ) {
-        let mut register = Register::new([state, state2, state3, state4, state5, state6, state7, state8]);
+        let mut register = Register::new(vec![state, state2, state3, state4, state5, state6, state7, state8]);
```
Comment (BAMF0) [inline tests/register_operation_tests.rs:61]: We should refactor these tests later as they can be dynamically sized now.
This does not need to be handled by this issue though.

File: src/register.rs (hunk 1 above)
Comment (simonengstrm) [reply src/register.rs]: Should be changed now in the latest commit

**Comment by BAMF0** [comment]
The build checks fail, they should be resolved before merge

**Review by BAMF0 — APPROVED** [review]
Looks good!

**Pull request merged**